src/Core/Product/Search/SortOrder.php

```diff
@@ -120,7 +120,13 @@ public function toString()
      */
     public static function newFromString($sortOrderConfiguration)
     {
-        list($entity, $field, $direction) = explode('.', $sortOrderConfiguration);
+        $sortParams = explode('.', $sortOrderConfiguration);
+
+        if (count($sortParams) < 3) {
+            throw new \PrestaShopException('Invalid argument');
```

Comment:
It should be an InvalidArgumentException or a CoreException. PrestaShopException is too generic.

Reply:
Done with a CoreException